src/decirest.erl

```diff
@@ -222,7 +174,17 @@ do_callback(Callback, Req, #{module := Module} = State, Default) ->
 do_callback(Module, Callback, Req, State, Default) ->
   case erlang:function_exported(Module, Callback, 2) of
     true ->
-      Module:Callback(Req, State);
+      case Module:Callback(Req, State) of
```

Comment:
this is an extra case so that the exported module can invoke the default behaviour but still modify the req and state object, correct?

Reply:
yes so that the resource doesn't do need to know the calling function.